webpack/scenes/ContentViews/Details/Filters/CVContainerImageFilterContent.js

```diff
@@ -0,0 +1,179 @@
+import getContentViewDetails, { getCVFilterRules, removeCVFilterRule } from '../ContentViewDetailActions';
+import AddEditContainerTagRuleModal from './Rules/ContainerTag/AddEditContainerTagRuleModal';
+
+const emptyContentTitle = __('No package groups have been added to this filter.');
+const emptyContentBody = __("Add to this filter using the 'Add package group' button.");
+const emptySearchTitle = __('No matching package groups found.');
+const emptySearchBody = __('Try changing your search settings.');
+
+const CVContainerImageFilterContent = ({ cvId, filterId }) => {
+  const dispatch = useDispatch();
+  const response = useSelector(state => selectCVFilterRules(state, filterId), shallowEqual);
+  const status = useSelector(state => selectCVFilterRulesStatus(state, filterId), shallowEqual);
+  const { results: filterResults } =
+    useSelector(state => selectCVFilters(state, cvId), shallowEqual);
+  const [rows, setRows] = useState([]);
+  const [metadata, setMetadata] = useState({});
+  const [searchQuery, updateSearchQuery] = useState('');
+  const [activeTabKey, setActiveTabKey] = useState(0);
+  const [selectedFilterRuleData, setSelectedFilterRuleData] = useState(0);
+  const [modalOpen, setModalOpen] = useState(false);
+  const loading = status === STATUS.PENDING;
+  const [bulkActionOpen, setBulkActionOpen] = useState(false);
+  const deselectAll = () => setRows(rows.map(row => ({ ...row, selected: false })));
+  const toggleBulkAction = () => setBulkActionOpen(prevState => !prevState);
+  const hasSelected = rows.some(({ selected }) => selected);
+
+  const onClose = () => {
+    setModalOpen(false);
+    setSelectedFilterRuleData(undefined);
+  };
+
+  const columnHeaders = [
+    __('Tag name'),
+  ];
+
+  const buildRows = useCallback((results) => {
+    const newRows = [];
+    const filterRules = filterResults.find(({ id }) => id === Number(filterId))?.rules || [];
+    results.forEach((containerRule) => {
+      const { name, id, ...rest } = containerRule;
+
+      newRows.push({
+        filterRuleId: filterRules?.find(({ uuid }) => uuid === rest.uuid)?.id,
+        cells: [{ title: name }],
+        name,
+        id,
+      });
+    });
+
+    return newRows.sort(({ added: addedA }, { added: addedB }) => {
+      if (addedA === addedB) return 0;
+      return addedA ? -1 : 1;
+    });
+  }, [filterId, filterResults]);
+
+
+  const bulkRemove = () => {
+    setBulkActionOpen(false);
+    deselectAll();
+  };
+
+  useDeepCompareEffect(() => {
+    const { results, ...meta } = response;
+    setMetadata(meta);
+    if (!loading && results) {
+      const newRows = buildRows(results);
+      setRows(newRows);
+    }
+  }, [response, loading, buildRows]);
+
+  const actionResolver = () => [
+    {
+      title: __('Remove'),
+      onClick: (_event, _rowId, { filterRuleId }) => {
+        dispatch(removeCVFilterRule(filterId, filterRuleId, () =>
+          dispatch(getContentViewDetails(cvId))));
```

Comment:
Can probably just use getCVFilterRules(filterId) to prevent page refresh.

Reply:
SMRT.